### ISSUE TITLE
Lazy loading of object data

### ISSUE THREAD

**Issue by mcmanuss8** [peeringdb/peeringdb#1367]
**Is your feature request related to a problem? Please describe.**
When pulling up an object page  (ex: https://www.peeringdb.com/ix/171 ) we get a single payload with all of the data for the ix - the html, the js, the css, and most importantly, the data for the IX, the networks in the IX, and the facilities the exchange is in.

Instead we should follow the more modern approach:

* Send a base page with the layout and all the dependent js, css, etc (using react components if we decide to use react)
* For each section, make a request for the data we care about. When that data has been received, render the appropriate table. 

**Who is affected by the problem?**
Everyone who uses the frontend of peeringdb

**What is the impact?**
We *should* see significant improvements to the *perceived* speed of the page, as the user's browser will start pulling in dependancies immediately while we wait for data fetching. Instead, we're bottlenecked on data fetching before we can make all the requests for various js, css, images, etc. 

While the total amount of time to load all the data might actually stay the same, the user will feel like the page is working faster as they'll see *some* of the data right away.

**Are there security concerns?**
Not that I can think of.

**Are there privacy concerns?**
Not that I can think of.

**Describe the solution you'd like**
See above.

**Do you think this feature will require a formal desig
<img width="951" alt="Screenshot 2023-04-05 at 11 01 30 AM" src="https://user-images.githubusercontent.com/5892502/230122946-9246f296-a2a5-4fc1-8270-233ea73ffb70.png">

<img width="943" alt="Screenshot 2023-04-05 at 11 05 12 AM" src="https://user-images.githubusercontent.com/5892502/230122939-fb6dd645-76e8-4fab-aa81-e692e504030e.png">
n?**
Yes - we should consider which JS libraries we use before tackling this. I have a bias towards React, but that's not the only solution here

**Describe alternatives you've considered**
Keep status quo is an option.

**Could this feature request need support from the Admin Committee?**
It should not 

**What is the proposed priority?**
It's good to do, but not the end of the world (we've lived this long without it)

**Provide a rationale for any/all of the above**
User performance

**Additional context**
See screenshots. We're waiting ~10seconds in a worst case (no cached files) before we start pulling in any dependencies. This is much worse when the user is far from the AWS region we're hosted in

**Comment by martinhannigan** [comment]
+1

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
From: mcmanuss8 ***@***.***>
Sent: Wednesday, April 5, 2023 11:06:23 AM
To: peeringdb/peeringdb ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [peeringdb/peeringdb] Lazy loading of object data (Issue #1367)


Is your feature request related to a problem? Please describe.
When pulling up an object page (ex: https://www.peeringdb.com/ix/171 ) we get a single payload with all of the data for the ix - the html, the js, the css, and most importantly, the data for the IX, the networks in the IX, and the facilities the exchange is in.

Instead we should follow the more modern approach:

  *   Send a base page with the layout and all the dependent js, css, etc (using react components if we decide to use react)
  *   For each section, make a request for the data we care about. When that data has been received, render the appropriate table.

Who is affected by the problem?
Everyone who uses the frontend of peeringdb

What is the impact?
We should see significant improvements to the perceived speed of the page, as the user's browser will start pulling in dependancies immediately while we wait for data fetching. Instead, we're bottlenecked on data fetching before we can make all the requests for various js, css, images, etc.

While the total amount of time to load all the data might actually stay the same, the user will feel like the page is working faster as they'll see some of the data right away.

Are there security concerns?
Not that I can think of.

Are there privacy concerns?
Not that I can think of.

Describe the solution you'd like
See above.

**Do you think this feature will require a formal desig
[Screenshot 2023-04-05 at 11 01 30 AM]<https://user-images.githubusercontent.com/5892502/230122946-9246f296-a2a5-4fc1-8270-233ea73ffb70.png>

[Screenshot 2023-04-05 at 11 05 12 AM]<https://user-images.githubusercontent.com/5892502/230122939-fb6dd645-76e8-4fab-aa81-e692e504030e.png>

n?** Yes - we should consider which JS libraries we use before tackling this. I have a bias towards React, but that's not the only solution here

Describe alternatives you've considered
Keep status quo is an option.

Could this feature request need support from the Admin Committee?
It should not

What is the proposed priority?
It's good to do, but not the end of the world (we've lived this long without it)

Provide a rationale for any/all of the above
User performance

Additional context
See screenshots. We're waiting ~10seconds in a worst case (no cached files) before we start pulling in any dependencies. This is much worse when the user is far from the AWS region we're hosted in

**Comment by arnoldnipper** [comment]
+1

**Comment by arnoldnipper** [comment]
Is this a duplicate of #954?

**Comment by mcmanuss8** [comment]
> Is this a duplicate of #954?

More or less, yes. But since 954 is closed and the latency can still be high...

**Comment by grizz** [comment]
I took this out of my #1225 report because it needs more looking into on how heavy of a lift it will be, it will likely be a fairy heavy one. It will also likely have huge latency benefits and leverage caching a lot better.

+1 for sure
